contracts/governance/src/governance.rs

```diff
@@ -187,31 +221,19 @@ impl GovernanceContract {
         address: PublicKey,
         value: u64,
     ) -> Result<(), Error> {
-        self.validate_seed(
+        self.assert_seed(
             iter::once([seed, BlsScalar::from(TX_BURN as u64)])
                 .chain(iter::once(address.as_ref().to_hash_inputs()))
                 .flatten()
                 .chain(iter::once(BlsScalar::from(value)))
                 .collect(),
             signature,
         )?;
-        self.check_pause()?;
 
+        self.assert_pause()?;
         self.total_supply = self.total_supply.saturating_sub(value);
 
-        let value = self
-            .balances
-            .get(&address)?
-            .copied()
-            .unwrap_or(0)
-            .checked_sub(value)
-            .ok_or(Error::InsufficientBalance)?;
-
-        if value == 0 {
-            self.balances.remove(&address)?;
-        } else {
-            self.balances.insert(address, value)?;
-        }
+        self.balance_sub(&address, value);
```

Comment:
Why the return value is not checked? It can contain some "remaining" value to subtract.

If this check is not needed, add a comment explaining why